File: backend/app/routes/organizations.py
"""
Organization routes — CRUD endpoints.
No logic yet; route stubs only.
"""
from fastapi import APIRouter

router = APIRouter()


@router.get("/", summary="List organizations")
async def list_organizations():
    raise NotImplementedError


@router.post("/", summary="Create organization", status_code=201)
async def create_organization():
    raise NotImplementedError


@router.get("/{org_id}", summary="Get organization by ID")
async def get_organization(org_id: int):
    raise NotImplementedError


@router.patch("/{org_id}", summary="Update organization")
async def update_organization(org_id: int):
    raise NotImplementedError


@router.delete("/{org_id}", summary="Delete organization", status_code=204)
async def delete_organization(org_id: int):
    raise NotImplementedError

File: frontend/src/api/organizations.ts
/**
 * Organizations API service — no logic yet, stubs only.
 */
export const organizationsApi = {
  list: () => {
    // TODO: GET /organizations
    throw new Error('Not implemented')
  },

  getById: (_id: number) => {
    // TODO: GET /organizations/:id
    throw new Error('Not implemented')
  },

  create: (_payload: unknown) => {
    // TODO: POST /organizations
    throw new Error('Not implemented')
  },

  update: (_id: number, _payload: unknown) => {
    // TODO: PATCH /organizations/:id
    throw new Error('Not implemented')
  },

  remove: (_id: number) => {
    // TODO: DELETE /organizations/:id
    throw new Error('Not implemented')
  },
}

export default organizationsApi
